In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")
%matplotlib inline
pd.options.display.max_columns = None
pd.options.display.float_format = '{:.2f}'.format

In [2]:
df_dirty = pd.read_csv(r'C:\\Users\\rashid\\Desktop\\customer_churn_prediction\\data\\raw\\WA_Fn-UseC_-Telco-Customer-Churn.csv')
df = df_dirty.copy()
df.head()

,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,OnlineBackup,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,Yes,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,No,Yes,No,No,No,One year,No,Mailed check,56.95,1889.5,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,Yes,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,No,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,No,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes


In [3]:
df.shape

(7043, 21)

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   str    
 1   gender            7043 non-null   str    
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   str    
 4   Dependents        7043 non-null   str    
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   str    
 7   MultipleLines     7043 non-null   str    
 8   InternetService   7043 non-null   str    
 9   OnlineSecurity    7043 non-null   str    
 10  OnlineBackup      7043 non-null   str    
 11  DeviceProtection  7043 non-null   str    
 12  TechSupport       7043 non-null   str    
 13  StreamingTV       7043 non-null   str    
 14  StreamingMovies   7043 non-null   str    
 15  Contract          7043 non-null   str    
 16  PaperlessBilling  7043 non-null   str    
 17  Paymen

## 📊 Data Dictionary (Məlumat Sütunları)

Layihədə istifadə olunan datada 7043 müştəriyə aid 21 sütun mövcuddur. Sütunların geniş təsviri aşağıdakı kəbdir:

| Sütun Adı | Data Tipi | Təsviri (Mənası) |
| :--- | :---: | :--- |
| **customerID** | str | Müştərinin unikal identifikasiya nömrəsi (ID) |
| **gender** | str | Müştərinin cinsi (Male/Female) |
| **SeniorCitizen** | int64 | Müştərinin təqaüdçü/yaşlı olub-olmaması (1: Bəli, 0: Xeyr) |
| **Partner** | str | Müştərinin partnyorunun olub-olmaması (Yes/No) |
| **Dependents** | str | Müştərinin himayəsində şəxslərin (uşaq, valideyn və s.) olub-olmaması (Yes/No) |
| **tenure** | int64 | Müştərinin şirkətdə qaldığı ayların sayı (Müştərilik müddəti) |
| **PhoneService** | str | Telefon xidmətinin aktiv olub-olmaması (Yes/No) |
| **MultipleLines** | str | Birdən çox telefon xəttinin olub-olmaması (Yes/No/No phone service) |
| **InternetService** | str | İnternet xidmət provayderinin növü (DSL, Fiber optic, No) |
| **OnlineSecurity** | str | Onlayn təhlükəsizlik xidmətinin olub-olmaması (Yes/No/No internet service) |
| **OnlineBackup** | str | Onlayn ehtiyat nüsxələmə (backup) xidmətinin olub-olmaması (Yes/No/No internet service) |
| **DeviceProtection** | str | Cihazın qorunması (sığortası) xidmətinin olub-olmaması (Yes/No/No internet service) |
| **TechSupport** | str | Texniki dəstək xidmətinin olub-olmaması (Yes/No/No internet service) |
| **StreamingTV** | str | TV yayımı xidmətinin olub-olmaması (Yes/No/No internet service) |
| **StreamingMovies** | str | Film yayımı xidmətinin olub-olmaması (Yes/No/No internet service) |
| **Contract** | str | Müqavilənin növü (Month-to-month, One year, Two year) |
| **PaperlessBilling** | str | Elektron faktura (kağızsız) sisteminin aktiv olub-olmaması (Yes/No) |
| **PaymentMethod** | str | Ödəniş üsulu (Electronic check, Mailed check, Bank transfer, Credit card) |
| **MonthlyCharges** | float64 | Müştərinin hər ay ödədiyi məbləğ |
| **TotalCharges** | str | Müştərinin ümumi ödədiyi məbləğ *(Qeyd: Bu sütun str tipindədir, float-a çevrilməlidir)* |
| **Churn** | str | **Hədəf Dəyişən (Target):** Müştərinin şirkəti tərk edib-etmədiyi (Yes/No) |

In [5]:
df['SeniorCitizen'].value_counts()

SeniorCitizen
0    5901
1    1142
Name: count, dtype: int64

In [6]:
df.describe()

,SeniorCitizen,tenure,MonthlyCharges
count,7043.00,7043.00,7043.00
mean,0.16,32.37,64.76
std,0.37,24.56,30.09
min,0.00,0.00,18.25
25%,0.00,9.00,35.50
50%,0.00,29.00,70.35
75%,0.00,55.00,89.85
max,1.00,72.00,118.75


In [7]:
df.isnull().sum()

customerID          0
gender              0
SeniorCitizen       0
Partner             0
Dependents          0
tenure              0
PhoneService        0
MultipleLines       0
InternetService     0
OnlineSecurity      0
OnlineBackup        0
DeviceProtection    0
TechSupport         0
StreamingTV         0
StreamingMovies     0
Contract            0
PaperlessBilling    0
PaymentMethod       0
MonthlyCharges      0
TotalCharges        0
Churn               0
dtype: int64

In [8]:
df['customerID'].nunique()

7043